product_multi_ean/models/product_product.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# © 2012-2014 Guewen Baconnier (Camptocamp SA)
+# © 2015 Roberto Lizana (Trey)
+# © 2016 Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+from openerp.addons.product import product as addons_product
+
+
+class ProductEan13(models.Model):
+    _name = 'product.ean13'
+    _description = "List of EAN13 for a product."
+    _order = 'sequence'
+
+    name = fields.Char(string='EAN13', size=13, required=True)
+    sequence = fields.Integer(string='Sequence', default=0)
+    product_id = fields.Many2one(
+        string='Product', comodel_name='product.product', required=True)
+
+    @api.multi
+    @api.constrains('name')
+    @api.onchange('name')
+    def _check_name(self):
+        for record in self:
+            if not addons_product.check_ean(record.name):
+                raise UserError(
+                    _('You provided an invalid "EAN13 Barcode" reference. You '
+                      'may use the "Internal Reference" field instead.'))
+
+    @api.multi
+    @api.constrains('name')
+    def _check_duplicates(self):
+        for record in self:
+            eans = self.search(
+                [('id', '!=', record.id), ('name', '=', record.name)])
+            if eans:
+                raise UserError(
+                    _('The EAN13 Barcode "%s" already exists for product '
+                      '"%s"!') % (record.name, eans[0].product_id.name))
```

Comment:
:-1: 
I'd argue being explicit here is an advantage. Imagine importing your barcodes from csv and receiving error message like "EAN13 Barcode already exists!". Then you'd have to manually find duplicate EANs in your file which is a bummer :-)